### PR TITLE
Allow the use of QuarkusTestResourceLifecycleManager with @TestProfile

### PR DESCRIPTION
**Pull request by geoand**

### CODE REVIEW

**Comment by geoand** [comment]
@stuartwdouglas let me know what you think. I am personally not real fond of the clunky signature of `testResources` so if you have any ideas, I'll love to hear them

File: test-framework/junit5/src/main/java/io/quarkus/test/junit/QuarkusTestExtension.java [hunk 1]
```diff
@@ -253,6 +256,21 @@ public void run() {
...
+            return Collections.emptyMap();
+        }
+
+        Map<Class<? extends QuarkusTestResourceLifecycleManager>, Map<String, String>> testResources = profileInstance
+                .testResources();
+        if (testResources.isEmpty()) {
+            return Collections.emptyMap();
+        }
+
+        return testResources;
```
Comment (gastaldi) [inline test-framework/junit5/src/main/java/io/quarkus/test/junit/QuarkusTestExtension.java:271]:
```suggestion
        return testResources;
```

**Comment by stuartwdouglas** [comment]
Maybe a custom type (TestResourceInstance?) that encapsulates the name and the parameters? The current signature does not allow for two resources of the same type with different params (e.g. two databases on different ports).

**Comment by geoand** [comment]
Good point

**Comment by geoand** [comment]
PR updated

**Review by stuartwdouglas — APPROVED** [review]

**Pull request merged**